### PR TITLE
Add tests for cidr, count, duration flags to hostfactory

### PR DESCRIPTION
**Pull request by sigalsax**
### What does this PR do?
Add UTs and integration tests for hostfactory 'create token' command

### What ticket does this PR close?
Resolves JIRA

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

**Review by orenbm — DISMISSED** [review]

File: test/test_integration_hostfactory.py [hunk 1]
```diff
@@ -49,3 +51,37 @@ def test_hostfactory_with_unknown_hostfactory_id_returns_error(self):
...
+
+        self.assertIn('[\n    {\n        "cidr": [\n            "1.2.0.0/16"\n        ],\n'
+                      f'        "expiration": "{(datetime.utcnow().replace(microsecond=0) + timedelta(hours=1)).isoformat()}Z",\n'
+                      '        "token":', output)
+
+    def test_hostfactory_wrong_cidr_format_raises_error(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory',
+                                  '--cidr', '1.2.3'], exit_code=1)
+        self.assertIn("Reason: 422", output)
```
Comment (eladkug) [inline test/test_integration_hostfactory.py:87]: new line is missing

File: test/test_unit_api.py [hunk 2]
```diff
@@ -43,7 +43,7 @@
 }
 
 
-MOCK_HOSTFACTORY_OBJECT = CreateTokenData(host_factory="some_host_factory")
+MOCK_HOSTFACTORY_OBJECT = CreateTokenData(host_factory="some_host_factory", cidr="1.2.3.4,0.0.0.0")
```
Comment (eladkug) [inline test/test_unit_api.py:46]: Please add the following tests:
1.2.3
1.2.3/16
1.2.3.4,1.2.3 (to verify if it put the right one only , or ignore entire list) 
without cdr (the one you removed) verify the default is working

**Review by eladkug — CHANGES_REQUESTED** [review]
Add some tests.
@szamir1  How can we delete cidr ? if we support it we need to test it
@szamir1 How many cidr we can write in one command, do we have limitation? we need to test it as well, if we dont have any declaration test with 10 different ip's

File: test/test_integration_hostfactory.py (hunk 1 above)
Comment (sigalsax) [reply test/test_integration_hostfactory.py]: done

File: test/test_unit_api.py (hunk 2 above)
Comment (sigalsax) [reply test/test_unit_api.py:46]: great thanks!

**Comment by szamir1** [comment]
> Add some tests. @szamir1 How can we delete cidr ? if we support it we need to test it @szamir1 How many cidr we can write in one command, do we have limitation? we need to test it as well, if we dont have any declaration test with 10 different ip's

@eladkug  @sigalsax  good questions! 
what do you mean by delete a cidr? once the restriction is given there is no ability to edit it. 

regarding the limitation - I think a limitation is a must, at least for this phase, as I believe it opens a place for problems.
Let's set the limitation to 10 CIDRs and add tests accordingly.

File: test/test_integration_hostfactory.py [hunk 3]
```diff
@@ -44,8 +54,136 @@ def test_hostfactory_without_id_returns_menu(self):
...
+    def test_hostfactory_with_only_hours_duration_flags_returns_correct_response(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory',
+                                  '--duration-hours', '24'])
+
+        self.assertIn('[\n    {\n        "cidr": [],\n'
+                      f'        "expiration": "{(datetime.utcnow().replace(microsecond=0) + timedelta(hours=24)).isoformat()}Z",\n'
+                      '        "token":', output)
+
+    def test_hostfactory_with_only_minutes_duration_flags_returns_correct_response(self):
```
Comment (eladkug) [inline test/test_integration_hostfactory.py:153]: add another tests with negative number for days for example

File: test/test_integration_hostfactory.py [hunk 4]
```diff
@@ -44,8 +54,141 @@ def test_hostfactory_without_id_returns_menu(self):
...
+                      '\n    {\n        "cidr": [],\n'
+                      f'        "expiration": "{(datetime.utcnow().replace(microsecond=0) + timedelta(hours=1)).isoformat()}Z",\n'
+                      f'        "token": "{token_values[2]}"\n    }}\n]\n'
+                      , output)
+
+    def test_hostfactory_with_negative_count_raises_error(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory', '--duration-minutes', '1'
+                                  '--count', '-1'], exit_code=1)
+        self.assertIn("Failed to execute command. Reason: Missing required", output)
```
Comment (eladkug) [inline test/test_integration_hostfactory.py:188]: Why not `self.assertIn("Failed to execute command. Reason: Parameter", output)`

File: test/test_integration_hostfactory.py [hunk 5]
```diff
@@ -44,8 +54,141 @@ def test_hostfactory_without_id_returns_menu(self):
...
+                      '        "token":', output)
+
+    def test_hostfactory_with_negative_duration_days_flags_raises_error(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory',
+                                  '--duration-days', '-1'], exit_code=1)
+
+        self.assertIn("Failed to execute command. Reason: Parameter", output)
+
+    def test_hostfactory_with_count_returns_correct_response(self):
```
Comment (eladkug) [inline test/test_integration_hostfactory.py:167]: Change the test to verify 10 instead of 3

File: test/test_integration_hostfactory.py [hunk 6]
```diff
@@ -44,8 +54,141 @@ def test_hostfactory_without_id_returns_menu(self):
...
+                                  '--cidr', '1.2.3/16'], exit_code=1)
+        self.assertIn("Reason: 422", output)
+
+    def test_hostfactory_with_valid_and_invalid_cidr_raises_error(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory', '--duration-hours', '1'
+                                  '--cidr', '1.2.3.4,1.2.3'], exit_code=1)
+        self.assertIn("Reason: 422", output)
+
+    def test_hostfactory_with_all_duration_flags_returns_correct_response(self):
```
Comment (eladkug) [inline test/test_integration_hostfactory.py:117]: need to add test without duration and get an error

**Review by eladkug — CHANGES_REQUESTED** [review]
Left some comments

File: test/test_integration_hostfactory.py (hunk 5 above)
Comment (sigalsax) [reply test/test_integration_hostfactory.py:174]: this isn't final so will add the logic and test when it is

File: test/test_integration_hostfactory.py (hunk 4 above)
Comment (sigalsax) [reply test/test_integration_hostfactory.py:195]: you would be correct!

File: test/test_integration_hostfactory.py [hunk 7]
```diff
@@ -44,8 +54,148 @@ def test_hostfactory_without_id_returns_menu(self):
...
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory', '--duration-minutes', '1',
+                                  '--count', '-1'], exit_code=1)
+        self.assertIn("Failed to execute command. Reason: Parameter", output)
+
+    def test_hostfactory_with_zero_count_raises_error(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory', '--duration-minutes', '1'
+                                  '--count', '0'], exit_code=1)
+        self.assertIn("Failed to execute command. Reason: Missing required", output)
```
Comment (eladkug) [inline test/test_integration_hostfactory.py:201]: new line is missing

File: test/test_integration_hostfactory.py [hunk 8]
```diff
@@ -44,8 +54,148 @@ def test_hostfactory_without_id_returns_menu(self):
...
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory',
+                                  '--duration-days', '-1'], exit_code=1)
+
+        self.assertIn("Failed to execute command. Reason: Parameter", output)
+
+    def test_hostfactory_without_duration_raises_error(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory',
+                                  '--duration-minutes', '1'], exit_code=1)
```
[the remark this answers is not in the document]
Comment (sigalsax) [reply test/test_integration_hostfactory.py]: shouldnt have added the duration. Nice catch
